packages/x-components/src/x-modules/related-prompts/components/related-prompts-list.vue

```diff
@@ -66,23 +69,48 @@
       showOnlyAfterOffset: {
         type: Boolean,
         default: false
-      }
+      },
+      customQuery: String
     },
     setup(props, { slots }) {
+      const x = use$x();
       const { query, status } = useState('relatedPrompts', ['query', 'status']);
 
       /**
        * The state related prompts.
        */
-      const relatedPrompts: ComputedRef<RelatedPrompt[]> = useState('relatedPrompts', [
-        'relatedPrompts'
-      ]).relatedPrompts;
+      const relatedPrompts: ComputedRef<Dictionary<RelatedPromptsItems>> = useState(
+        'relatedPrompts',
+        ['relatedPrompts']
+      ).relatedPrompts;
 
       /**
        * Injected query, updated when the related request(s) have succeeded.
        */
       const injectedQuery = inject<Ref<string | undefined>>(QUERY_KEY as string);
 
+      const relatedPromptsProducts: Ref<RelatedPrompt[]> = ref([]);
+
+      watch(
+        () => props.customQuery,
+        () => {
+          if (props.customQuery || props.customQuery !== '') {
+            x.emit('RelatedPromptsCustomQueryProvider', props.customQuery);
```

Comment:
Following the comment above, I'd like to skip the custom prefix. Besides, we should stick to our convention of using past tense. What about `RelatedPromptsQueryProvided`?